### ISSUE TITLE
Proposal: add GetEnvironmentVariableAsBool to netstandard

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#26079]
@diddledan commented on [Tue Dec 19 2017](https://github.com/dotnet/standard/issues/608)

Currently the dotnet cli has a function which is located at: [src/Microsoft.DotNet.Cli.Utils/EnvironmentProvider.cs Line 106](https://github.com/dotnet/cli/blob/fffe3142ad8043743d83885df2166fd2f80dceca/src/Microsoft.DotNet.Cli.Utils/EnvironmentProvider.cs#L106)

I think it would be useful to consider this functionality for inclusion into the `System.Environment` namespace.

This issue is to raise the idea and garner discussion. I do not believe it would conflict with anything already present, though I've not performed a thorough examination to prove it.

---

@danmosemsft commented on [Wed Dec 27 2017](https://github.com/dotnet/standard/issues/608#issuecomment-354183176)

@diddledan the right place to propose new API is the corefx repo...

**Comment by diddledan** [comment]
thanks for moving this to the correct place, @danmosemsft :-)

Some thoughts for rationale:
I'm looking at powershell on Linux systems which could benefit from this or a similar function which is currently only available via copy+paste from dotnet cli or via reimplementation from scratch. As this seems to be a potentially beneficial API for the wider community I think it worthy of including as an additional function in `System.Environment` so that projects may rely on it without rolling their own reinvention (or creating their own mixed-metaphors to labour the point :-p). It's also a very simple standalone function which shouldn't knock-on in other areas.

Specific to my intentions for powershell, I want to add an environment variable check to [Microsoft.PowerShell.ConsoleHost:Telemetry.cs](https://github.com/PowerShell/PowerShell/blob/c17a975683a0df639030521777d9cdc764784799/src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs#L48) similar to the one in [dotnet-cli](https://github.com/dotnet/cli/blob/00bf9f9f6dc7c65699ebb1a32217ca496f4c0251/src/dotnet/Telemetry/Telemetry.cs#L34) that allows for runtime disabling of telemetry gathering without requiring the filesystem to be modified.

**Comment by danmosemsft** [comment]
@diddledan sounds good, can you please reformat with sections as suggested in https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md 

That would organize the proposed API in C# format, suggestion of the implementation, rationale/why it's important enough and folks can't roll their own.

**Comment by danmosemsft** [comment]
BTW you mentioned adding to netstandard. .NET Standard 2.0 is shipped and fixed. Because .NET Standard defines a common subset of API's across various .NET platforms, for an API to get into a future .NET Standard version it must exist in those various platforms first. The first one to start with is CoreFX.

**Comment by karelz** [comment]
Note: We don't even have a method to parse environment value as an integer.

The proposal above adds parsing logic and several variants of possible true/false encodings into .NET Core. I don't think it is a good idea.
The values are not localized and it feels weird that we should decide in the platform how true/false can be encoded (it is not as straightforward as integer encoding and involves much more language-specific text/characters).

What might help IMO: Demonstrate there is non-trivial number of projects implementing such function on their own today. (e.g. search GitHub)

**Comment by danmosemsft** [comment]
I agree with @karelz : what does this add over 2 lines: GetEnvironmentVariable and bool.TryParse.

**Comment by diddledan** [comment]
I've added some search results to the post indicating projects that I found from a quick browse of Github. The first 5 pages of the 100 pages of results from the first search (Terms: GetEnvironmentVariable bool; Filtered to only c# files) yeilded the second list of items:

[Dotnet CLI](https://github.com/dotnet/cli/blob/fffe3142ad8043743d83885df2166fd2f80dceca/src/Microsoft.DotNet.Cli.Utils/EnvironmentProvider.cs#L106); [Microsoft aspnet sdk](https://github.com/aspnet/websdk/blob/d933fdfb097a37ab5eefa85ebfe09246fa9a2216/src/Publish/Microsoft.NET.Sdk.Publish.Tasks/EnvironmentHelper.cs#L12); [Microsoft Azure functions CLI](https://github.com/Azure/azure-functions-cli/blob/ec6b327df40cfcfa7269ea13c8bffcd650419eaf/src/Azure.Functions.Cli/Common/StaticSettings.cs#L7); [Microsoft Xamarin Forms](https://github.com/xamarin/Xamarin.Forms/blob/d078ba64da3a844d266a28a849690355434e7021/Xamarin.Forms.ControlGallery.Android/TestCloudService.cs#L7); [1](https://github.com/csMACnz/coveralls.net/blob/931979347e8071bf9e873f3f514b25801a626b97/src/csmacnz.Coveralls/AppVeyorGitDataResolver.cs#L14); [2](https://github.com/dotnetsheff/dotnetsheff-api/blob/185c59dcf84bd4d535add65b0a8d1fa53c15aad6/src/dotnetsheff.Api/AlexaSkill/AlexaSkillSettings.cs#L9); [3](https://github.com/shouldly/shouldly/blob/cb58830180c1c3f87db9b0eb74ef9d7c446ec71e/src/Shouldly/Configuration/DoNotLaunchWhenEnvVariableIsPresent.cs#L15); [4](https://github.com/zopyx/zopyx.csharp.smashdocs/blob/e476656f97c7f9e41ae65e29cc08a5357ae784ff/test/EmptyStruct.cs#L12); [5](https://github.com/cjac/deb-dotnet-corefx/blob/b92e6afd5fa7c4b3a41a634df7359ade63c27580/src/Common/tests/System/Net/TestSettings.cs#L63); [6](https://github.com/JacobMisirian/Hassium-Rewrite/blob/d40f54d617c55a57c8189e8c856aae0c9eac7034/src/Hassium/Runtime/Net/HassiumCGI.cs#L141); [7](https://github.com/ringcentral/ringcentral-csharp-client/blob/90f545e7a9b07e024f34877e8ce71059755a567c/RingCentral.Test/Config.cs#L25); [8](https://github.com/gitsandy2016/TESTGIT/blob/e3504215ab2ae910f960902db3b5d7774720c420/WebsiteENvironment.cs#L26)

**Comment by justinvp** [comment]
This might be a good candidate for a standalone helper API outside of corefx (corefxlab or 3rd party nuget package) that can be used to parse boolean strings in this way from any source (e.g. environment variables, command line args, windows registry, etc.).
